sml/src/sml_transport.c

```diff
@@ -34,15 +34,16 @@ unsigned char end_seq[] = {0x1b, 0x1b, 0x1b, 0x1b, 0x1a};
 
 
 size_t sml_read(int fd, fd_set *set, unsigned char *buffer, size_t len) {
-	
+
 	ssize_t r = 0;
 	size_t tr = 0;
 
 	while (tr < len) {
 		select(fd + 1, set, 0, 0, 0);
 		if (FD_ISSET(fd, set)) {
-			
+
 			r = read(fd, &(buffer[tr]), len - tr);
+			if (r == 0) return 0;
 			if (r < 0) continue;
```

Comment:
this will still loop forever on read errors,
errors will never be passed to the caller.
( in reply to https://github.com/volkszaehler/libsml/pull/43#issuecomment-360503305 )

Reply:
imho we should explicitly check for EINTR and EGAIN and abort on other errors...

Reply:
Errors wont be passed anyway unless we change signature or add some error variable?

Reply:
it seems silly to continue forever on fatal errors.
i'd log to stderr and return 0.
as (within the possibilities of the current API).